deps/verifier/src/snp/mod.rs

```diff
@@ -44,57 +42,95 @@ const UCODE_SPL_OID: Oid<'static> = oid!(1.3.6 .1 .4 .1 .3704 .1 .3 .8);
-pub(crate) fn load_milan_cert_chain() -> &'static Result<VendorCertificates> {
-    static MILAN_CERT_CHAIN: OnceLock<Result<VendorCertificates>> = OnceLock::new();
-    MILAN_CERT_CHAIN.get_or_init(|| {
-        let certs = X509::stack_from_pem(include_bytes!("milan_ask_ark_asvk.pem"))?;
+pub(crate) fn load_cert_chain(proc_model: ProcType) -> &'static Result<VendorCertificates> {
+    static CERT_CHAIN: OnceLock<Result<VendorCertificates>> = OnceLock::new();
+    CERT_CHAIN.get_or_init(|| {
+        let cert_authority = match proc_model {
+            ProcType::Milan => include_bytes!("milan_ask_ark_asvk.pem"),
+            ProcType::Genoa => include_bytes!("genoa_ask_ark_asvk.pem"),
+            ProcType::Turin => include_bytes!("turin_ask_ark_asvk.pem"),
+        };
+
+        let certs = X509::stack_from_pem(cert_authority).context("Failed to parse PEM file")?;
+
         if certs.len() != 3 {
-            bail!("Malformed Milan ASK/ARK/ASVK");
+            return Err(anyhow!(
+                "Malformed cached Vendor Certs for {} processor (ASK, ARK, ASVK)",
+                proc_model
+            ));
         }
 
         let vendor_certs = VendorCertificates {
-            ask: certs[0].clone(),
-            ark: certs[1].clone(),
-            asvk: certs[2].clone(),
+            ask: Certificate::from(certs[0].clone()),
+            ark: Certificate::from(certs[1].clone()),
+            asvk: Certificate::from(certs[2].clone()),
         };
+
         Ok(vendor_certs)
     })
 }
 
-impl Snp {
-    /// Creates a new `Snp` instance by loading the Milan certificate chain.
-    /// Returns an error if the certificate chain can not be loaded.
-    pub fn new() -> Result<Self> {
-        let Result::Ok(vendor_certs) = load_milan_cert_chain() else {
-            bail!("Failed to load Milan cert chain");
-        };
-        let vendor_certs = vendor_certs.clone();
-        Ok(Self { vendor_certs })
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) enum ProcType {
+    /// 3rd Gen AMD EPYC Processor (Standard)
+    Milan,
+
+    /// 4th Gen AMD EPYC Processor (Standard)
+    Genoa,
+
+    /// 5th Gen AMD EPYC Processor (Standard)
+    Turin,
+}
+
+impl FromStr for ProcType {
+    type Err = anyhow::Error;
+    fn from_str(input: &str) -> Result<ProcType, anyhow::Error> {
+        match input.to_lowercase().as_str() {
+            "milan" => Ok(ProcType::Milan),
+            "genoa" => Ok(ProcType::Genoa),
+            "turin" => Ok(ProcType::Turin),
+            _ => Err(anyhow::anyhow!("Processor type not found!")),
+        }
     }
 }
 
-#[derive(Clone, Debug)]
-pub(crate) struct VendorCertificates {
-    ask: X509,
-    ark: X509,
-    asvk: X509,
+impl fmt::Display for ProcType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            ProcType::Milan => write!(f, "Milan"),
+            ProcType::Genoa => write!(f, "Genoa"),
+            ProcType::Turin => write!(f, "Turin"),
+        }
+    }
```

Comment:
we can use [strum](https://docs.rs/strum_macros/latest/strum_macros/derive.Display.html) to handle this